### ISSUE TITLE
Canceling is not hooked up for jobs in containers

### ISSUE THREAD

**Issue by AlanCoding** [ansible/ansible-runner#506]
I am basing my example here off the `ansible-builder` pytz example:

https://github.com/ansible/ansible-builder/tree/devel/test/data/pytz

modify the `pytz.yml` playbook to sleep for 1 minute:

```yaml
---
- hosts: localhost
  gather_facts: false
  connection: local
  tasks:
  - name: Debug using the rrule plugin
    debug:
      msg: >
        {{ query(
          'awx.awx.tower_schedule_rrule',
          'minute',
          start_date='2020-4-16 03:45:07',
          end_on='2'
        ) }}

  - name: sleep
    command: sleep 60

```

Build your image an all that jazz, and run this for a reference:

```
ansible-runner run test/data/pytz --playbook=pytz.yml
```

Watch it finish.
Check `docker ps -a` and you find that the container is now gone (expected).

Now, run this bash script:

```
$ cat alan.sh 
ansible-runner start test/data/pytz --playbook=pytz.yml
echo $?
ansible-runner stop test/data/pytz
echo $?
```

Expected result: no running container after this finishes.

Actual result: running container persists after this finishes for roughly 1 minute until the process has finished.

In practice, I have experienced several jobs where they are completely blocked on something and should be canceled. Because canceling doesn't do anything but reports no error code, the containers are never cleaned up.

**Comment by AlanCoding** [comment]
should we use `docker kill`?

https://docs.docker.com/engine/reference/commandline/kill/

I also have a really big question of reference-ability. How do we identify the container that needs to be killed?

**Comment by AlanCoding** [comment]
The fix for this has been merged in #530
